packages/ui/src/machines/authenticator/defaultServices.ts

```diff
@@ -41,6 +41,8 @@ export const defaultServices = {
     };
   },
   async getCurrentUser() {
+    console.log('+++getCurrentUser');
+    await new Promise((resolve) => setTimeout(resolve, 4));
```

Comment:
- There is a race condition between the machine being reinitialized after redirect and the JS API logging in the user
- If the user attempts to call an authenticated API in this time it seems to hang here but adding this short wait resolves this issue.
- This seems to not occur on react because on the signedIn event react re-renders whereas this does not seem to occur in angular - https://github.com/aws-amplify/amplify-ui/blob/next-release/migration/packages/ui/src/helpers/authenticator/utils.ts#L85

Reply:
Is this an issue in the current Authenticator or just with the migration?

Reply:
I'm not able to get the angular example app for v5 working with federated sign in, I consistently get an error and instpecting the network log for the 400 it shows

Partially omitted the error for saftey
```
{"__type":"ValidationException","message":"1 validation error detected: Value '{cognito-idp.us-east-2.amazonaws.com/us-east-2********}' at 'logins' failed to satisfy constraint: Map value must satisfy constraint: [Member must have length less than or equal to 50000, Member must have length greater than or equal to 1]"}
```

Reply:
What is the impact of removing the call to `getCurrentUser` after redirect?

Reply:
This call to `getCurrentUser` happens everytime the state machine is instantiated - https://github.com/aws-amplify/amplify-ui/blob/next-release/fix-vue-federated/packages/ui/src/machines/authenticator/index.ts#L70

It's how we check for whether the user is authenticated or not so removing it here might be complicated to say the least

Reply:
Would it be possible not to call it on redirect though?

Reply:
Ended up attempting something else which might be the fix